portfolio/src/main/webapp/js/comments.js

```diff
@@ -170,14 +174,31 @@ function addCommentToPage(comment) {
     deleteButton.innerText = "Delete";
     deleteButton.addEventListener(
         "click", 
-        createCommentDeleter(comment.id, newComment));
+        createCommentDeleter(comment.id, commentRow));
 
     authorFooter.appendChild(deleteButton);
   }
 
   newComment.appendChild(authorFooter);
 
-  commentContainer.appendChild(newComment);
+  if (comment.attachedImageUrl) {
+    const attachedImage = document.createElement("img");
+    attachedImage.src = comment.attachedImageUrl;
+    attachedImage.classList.add("img-thumbnail", "comment-image", "my-2");
+
+    const imageColumn = document.createElement("div");
+    imageColumn.classList.add("col-sm-3", "d-flex", "justify-content-center");
+    imageColumn.appendChild(attachedImage);
+
+    const commentColumn = document.createElement("div");
+    commentColumn.classList.add("col-sm-9");
+    commentColumn.appendChild(newComment);
+
+    commentRow.appendChild(imageColumn);
```

Comment:
The function `addCommentToPage` is getting pretty long so I think this image-adding logic should go in a separate function that gets called from within `addCommentToPage`

Reply:
Totally agree. Separated it into more focused functions. Thanks!